plugins/outputs/elasticsearch/elasticsearch_test.go

```diff
@@ -257,3 +259,70 @@ func TestGetIndexName(t *testing.T) {
 		}
 	}
 }
+
+func TestRequestHeaderWhenGzipIsEnabled(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.URL.Path {
+		case "/_bulk":
+			require.Equal(t, "gzip", r.Header.Get("Content-Encoding"))
+			require.Equal(t, "gzip", r.Header.Get("Accept-Encoding"))
+			w.WriteHeader(http.StatusOK)
+			w.Write([]byte("{}"))
```

Comment:
I think this will clear up the linter errors about not checking the return favlue of w.Write().  You'll need to do the same thing in the three other places in this file where the linter gives an error.

```suggestion
			require.NoError(t, w.Write([]byte("{}")))
```